### PR TITLE
fix(GraphQL): RHICOMPL-3244 make tags a named scope for systems

### PR DESCRIPTION
**Pull request by skateman**
Should fix the problem of non-scoped tags, depends on https://github.com/RedHatInsights/compliance-backend/pull/1437

## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [x] General Coding Practices

### CODE REVIEW

**Review by romanblanco — APPROVED** [review]
LGTM :+1:

**Comment by skateman** [comment]
@vkrizan wdyt about this one? We're storing a scope from before applying the tagging in order to list all available tags later.

File: app/graphql/resolvers/concerns/collection.rb [hunk 1]
```diff
@@ -93,6 +95,16 @@ def authorized_scope
...
       end
+
+      # Store the scope without applying any filtering by tags on it. This is
+      # required for listing the available tags in a view without any influence
+      # of the tagging scope. The selection is based on the second argument of
+      # the lambdas defined by each filter.
+      def store_wide_scope!(filter = ->(_, nil: nil) {})
+        return yield if %i[_tags_filter _pagination_filter].include?(filter.parameters[1][1])
+
+        context[:wide_scope] = yield
```
Comment (vkrizan) [inline app/graphql/resolvers/concerns/collection.rb:106]: Looks like the filters are not stored, so why not just store the `base_scope`? What am I missing?

This implementation seems very rigid. Can you think of a simplified one that can be better understandable?

**Review by vkrizan — CHANGES_REQUESTED** [review]

**Review by vkrizan — APPROVED** [review]
Now we're talking. :+1:

**Pull request merged**